cmd/operator-sdk/bundle/add.go

```diff
@@ -0,0 +1,227 @@
+
+More information on operator index images:
+https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-registry.md
+More information on 'opm':
+https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md
+`,
+			registryVersion),
+		Example: `The following invocation will create a new test-operator bundle index image:
+
+  $ operator-sdk bundle add quay.io/example/test-operator:v0.1.0 \
+      --to-index quay.io/example/test-operator-index:v0.1.0
+
+The following invocation will add a test-operator bundle image to an existing
+index image at version v0.1.0, creating a new index image at version v0.2.0:
+
+  $ operator-sdk bundle add quay.io/example/test-operator:v0.2.0 \
+      --from-index quay.io/example/test-operator-index:v0.1.0 \
+      --to-index quay.io/example/test-operator-index:v0.2.0
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("a comma-separated list of bundle image tags is a required argument, " +
+					"ex. 'example.com/test-operator:v0.1.0,example.com/test-operator:v0.2.0'")
+			}
+			c.bundlesImages = args[0]
+
+			if err := c.validate(); err != nil {
+				return fmt.Errorf("error validating args: %v", err)
+			}
+
+			binaryPath := filepath.Join("bin", "opm")
+			if runtime.GOOS == "windows" {
+				binaryPath += ".exe"
+			}
+			if err := c.buildOPM(binaryPath); err != nil {
+				log.Fatalf("Error building image builder: %v", err)
+			}
+			if err := c.runOPMIndexAdd(binaryPath); err != nil {
+				log.Fatalf("Error buildingindex image: %v", err)
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&c.fromIndex, "from-index", "f", "", "Previous index to build new index image from")
+	cmd.Flags().StringVarP(&c.toIndex, "to-index", "t", "", "Tag for new index image being built")
+	if err := cmd.MarkFlagRequired("to-index"); err != nil {
+		panic(err)
+	}
+	cmd.Flags().StringVar(&c.dockerfileName, "dockerfile-name", "",
+		"Name of the Dockerfile to generate if --generate-only is set. Default is 'Dockerfile'")
+	cmd.Flags().StringVar(&c.imageBuilder, "image-builder", "docker",
+		"Tool to build container images. One of: [docker, podman]")
+	cmd.Flags().BoolVarP(&c.generateOnly, "generate-only", "g", false,
+		"Generate metadata and a Dockerfile on disk without building the bundle image")
+	cmd.Flags().BoolVar(&c.permissive, "permissive", false,
+		"Allow registry load errors without exiting the build")
+
+	return cmd
+}
+
+func (c bundleIndexCmd) validate() error {
+	if !c.generateOnly && c.dockerfileName != "" {
+		return fmt.Errorf("dockerfile name can only be set if generating files")
+	}
+	return nil
+}
+
+// buildOPM reates and download 'opm' to a binary directory using 'go get' to
+// download a particular version, specified by registryVersion, to install.
+// TODO(estroz): shell out to make after kubebuilder integration.
+func (c bundleIndexCmd) buildOPM(binaryPath string) error {
```

Comment:
@ecordell is in the process of creating a downstream build of `opm`. Does it make more sense for the registry project to just include a release of the `opm` binary that could be downloaded -- or perhaps even just a container image to extract it from like `opm` itself does to do the container image build -- rather than compiling from source?

Trying to avoid requiring golang as a dependency if possible.